src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -395,9 +397,29 @@ static bool read_channel_args(grpc_chttp2_transport* t,
     }
   }
   if (channelz_enabled) {
+    char* host = nullptr;
+    int port_num = -1;
+    // try to pick out just the host port (maybe trims off scheme prefix).
+    grpc_uri* uri = grpc_uri_parse(t->peer_string, false);
+    // if peer string was a valid URI, we can use our lib to do the trimming.
+    if (uri != nullptr) {
+      const char* host_port = uri->path;
+      if (*host_port == '/') ++host_port;
+      char* port;
+      GPR_ASSERT(gpr_split_host_port(host_port, &host, &port));
+      if (port != nullptr) {
+        port_num = atoi(port);
+      }
+      gpr_free(port);
+    } else {
+      // if peer string is not a valid URI, just use the entire string to
```

Comment:
When will this happen?  Are we sure it's appropriate to report this as an IP address if it does, given that IP addresses should always be IPv4 or IPv6 URIs?  In particular, I'm thinking about cases where we create a channel from an application-provided fd.

Now that I think about it, we also need to handle unix domain sockets, which should be URIs as well.